src/UnderlineNav2/UnderlineNavItem.tsx

```diff
@@ -70,30 +77,73 @@ export const UnderlineNavLink = forwardRef(
     const textStyles: BetterSystemStyleObject = {
       whiteSpace: 'nowrap'
     }
+
+    const wrapperStyles = {
+      display: 'inline-flex',
+      paddingY: 1,
+      paddingX: 2,
+      borderRadius: 2
+    }
     const smallVariantLinkStyles = {
-      paddingY: 2,
+      paddingY: 1,
       fontSize: 0
     }
     const defaultVariantLinkStyles = {
-      paddingY: 3,
+      paddingY: 2,
       fontSize: 1
     }
 
     const linkStyles = {
+      position: 'relative',
       display: 'inline-flex',
       color: 'fg.default',
       textAlign: 'center',
-      borderBottom: '2px solid transparent',
-      borderColor: selectedLink === ref ? 'primer.border.active' : 'transparent',
       textDecoration: 'none',
-      paddingX: 2,
-      marginRight: 3,
+      paddingX: 1,
+      borderColor: selectedLink === ref ? 'primer.border.active' : 'transparent',
       ...(variant === 'small' ? smallVariantLinkStyles : defaultVariantLinkStyles),
-      '&:hover, &:focus': {
-        borderColor: selectedLink === ref ? 'primer.border.active' : 'neutral.muted',
-        transition: '0.2s ease'
+      '&:hover > div[data-component="wrapper"] ': {
+        backgroundColor: 'neutral.muted',
+        transition: 'background .12s ease-out'
+      },
+      '&:focus': {
```

Comment:
new focus implementation. As @langermank pointed out today that Safari doesn't respect borderRadius for the outlines so new focus implementation, inspired from [Primer CSS](https://github.com/primer/css/blob/main/src/support/mixins/misc.scss)